common/src/main/java/com/netflix/conductor/common/metadata/workflow/SubWorkflowParams.java

```diff
@@ -100,31 +94,28 @@ public Object getWorkflowDefinition() {
         return workflowDefinition;
     }
 
-    /**
-     * @return the workflowDefinition as a WorkflowDef
-     */
-    @JsonGetter("workflowDefinition")
-    public WorkflowDef getWorkflowDef() {
-        return (WorkflowDef) workflowDefinition;
-    }
-
     /**
      * @param workflowDef the workflowDefinition to set
      */
     public void setWorkflowDefinition(Object workflowDef) {
-        if (!(workflowDef == null || workflowDef instanceof WorkflowDef)) {
+        if (workflowDef == null) {
+            this.workflowDefinition = workflowDef;
+        } else if (workflowDef instanceof WorkflowDef) {
+            this.workflowDefinition = workflowDef;
+        } else if (workflowDef instanceof String) {
+            if (!(((String) workflowDef).startsWith("${"))
+                    || !(((String) workflowDef).endsWith("}"))) {
+                throw new IllegalArgumentException(
+                        "workflowDefinition is a string, but not a valid DSL string");
+            } else {
+                this.workflowDefinition = workflowDef;
+            }
+        } else if (workflowDef instanceof LinkedHashMap) {
+            this.workflowDefinition = TaskUtils.convertToWorkflowDef(workflowDef);
+        } else {
             throw new IllegalArgumentException(
-                    "workflowDefinition must be either null or WorkflowDef");
+                    "workflowDefinition must be either null, or WorkflowDef, or a valid DSL string");
```

Comment:
Shouldn't you add  `or a LinkedHashMap` in this exception?

Reply:
This exception will propagate to UI on workflow save. If I add LinkedHashMap It will only confuse users and will not tell them anything useful, because for the users workflowDefinition **cannot** be just a LinkedHashMap.